app/views/locator_maps/index.html.erb

```diff
@@ -0,0 +1,90 @@
+        </div>
+      </div>
+
+      <div class="usa-media_block">
+        <%= image_tag 'locator/texticon_scales.svg', alt: 'Weight Scales', class: 'usa-media_block-img' %>
+        <div class="usa-media_block-body">
+          <strong>Truck Weight Scales</strong> are privately owned locations where you can take your vehicles to be weighed to receive required Personally Procured (PPM) “Do-It-Yourself” move weight tickets.
+        </div>
+      </div>
+    </div>
+
+    <div class="location-search-section">
+      <div id="location-search-mine" class="location-search-mine">
+        <button id="button-search-mine" onclick="getLocation(this)">Search My Location</button>or
+      </div>
+
+      <%= form_with class: 'usa-search usa-search-big location-search-zipcode', method: 'get' do |form| %>
+        <div role="search">
+          <%= form.label :zipcode, "Search by ZIP code", class: 'usa-sr-only' %>
+          <%= form.text_field :zipcode, type: 'search', placeholder: 'Search by ZIP code' %>
+          <%= form.submit "Search" %>
+        </div>
+      <% end %>
+
+    </div>
+
+    <div id="location-error" class="usa-alert usa-alert-error" role="alert" hidden>
+      <div class="usa-alert-body">
+        <h3 class="usa-alert-heading">Location Error</h3>
+        <p class="usa-alert-text" id="location-error-body">Could not get your location.</p>
+      </div>
+    </div>
+
+    <div id="mapid"
+      data-origin-lat="<%= @origin[0] %>"
+      data-origin-lng="<%= @origin[1] %>"
+      data-marker-icon-path="<%= asset_path 'locator/map_pppo.svg' %>"
+      data-marker-icon-2x-path="<%= asset_path 'locator/map_pppo.svg' %>"
+      data-marker-shadow-path="<%= asset_path 'leaflet/dist/images/marker-shadow.png' %>"
+      data-spinner-path="<%= asset_path 'locator/ajax-loader.gif' %>">
+    </div>
+
+    <%- if @offices %>
+      <div id="pppo-search-results">
+        <%- @offices.each do |office| -%>
+          <%= render 'locator_maps/offices', office: office %>
+        <%- end %>
+        <%- if @offset + 5 < TransportationOffice.count %>
+          <%- if params[:zipcode].present? %>
+            <%= link_to 'Next 5 results', locator_maps_path(offset: @offset + 5, zipcode: params[:zipcode], anchor: 'mapid') %>
+          <%- elsif params[:coords].present? %>
+            <%= link_to 'Next 5 results', locator_maps_path(offset: @offset + 5, coords: params[:coords], anchor: 'mapid') %>
+          <%- end %>
+        <%- end %>
+      </div>
+    <%- end -%>
+  </div>
+</div>
+
+<script>
+  // simplest way to get the marker information over to the JS side
+
+  var markers = [];
+  var coordinates = [];
+  var locations = [];
+
+  <%- @offices.each do |office| %>
+    locations.push({
+      'id': <%= office.id %>,
+      'name': '<%= office.name %>',
+      'lat': <%= office.lat %>,
+      'lng': <%= office.lng %>
+    });
+  <%- end %>
```

Comment:
The goal here is to generate a static array, `locations`, so you shouldn't have to rely on `Array.push()` to get what you want. You could rewrite this block as:

```erb
<script>
  var locations = [
    <%- @offices.each do |office| -%>
      {
        id: <%= office.id %>,
        name: '<%= office.name %>',
        lat: <%= office.lat %>,
        lng: <%= office.lng %>
      },
    <%- end ->
  ];
</script>
```

I'm pretty sure the trailing comma would be okay, but you could also check if the loop is on the last iteration:

```erb
}<%= ',' unless office == @offices.last %>
```